libs/api/ebpf_api.cpp

```diff
@@ -1753,11 +1753,17 @@ _initialize_ebpf_object_from_native_file(
     }
 
 Exit:
-    ebpf_free(program);
     if (result != EBPF_SUCCESS) {
+        if (program && object.programs.size() == 0) {
```

Comment:
When a program gets added to `object.programs` the local variable `program` is set to `nullptr`. So, I think the `if (program && object.programs.size() == 0)` check is redundant IMO. Additionally, you may want to wrap the three free statements into a single function if one does not exist already.